src/java.desktop/macosx/native/libawt_lwawt/awt/AWTWindow.m

```diff
@@ -654,7 +654,9 @@ - (void) _deliverMoveResizeEvent {
     CHECK_EXCEPTION();
     (*env)->DeleteLocalRef(env, platformWindow);
 
-    [AWTWindow synthesizeMouseEnteredExitedEventsForAllWindows];
+    if (![self.nsWindow inLiveResize]) {
+        [AWTWindow synthesizeMouseEnteredExitedEventsForAllWindows];
+    }
```

Comment:
Are you sure that inLiveResize is only set when the user resizes the window? What happens if the user will move the window to the fullscreen or maximize it(by clicking on the title/buttons), will the inLiveResize be set to true or not?

Reply:
The docs seem to me to be saying it is only when the user is dragging :
https://developer.apple.com/documentation/appkit/nsview/1483267-inliveresize?language=objc

Reply:
When the user maximizes the window a mouseEntered event does fire, so it preserves the fix for JDK-8028485. But I noticed another problem is that a mouseEntered event will be generated if the window is dragged around (without resizing or entering the actual window). This could cause a situation where you can drag a window around to create a mouseEntered event, move the mouse away from the window, and re-enter the window without generating a mouseEntered event.